File: src/Python/429.TreeLevelOrder.py
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Node
        :rtype: List[List[int]]
        """
        r = []
        if root is None:
            return r
        stack = []
        stack.append(root)
        while len(stack) != 0:
            child = []
            n = len(stack)
            for i in range(n):
                node = stack.pop(0)
                child.append(node.val)
                if node.children is not None:
                    for c in node.children:
                        stack.append(c)
            r.append(child)
        return r

if __name__ == '__main__':
    solution = Solution()
    root = Node(1, [])
    node = Node(2, [])
    node.children.append(Node(3, []))
    root.children.append(node)
    root.children.append(node)
    print(solution.levelOrder(root))
else:
    pass

File: src/CSharp/429.TreeLevelOrder.cs

using System;
using System.Collections.Generic;

namespace PeefyLeetCode.TreeLevelOrder {


public class Node {
    public int val;
    public IList<Node> children;

    public Node(){}
    public Node(int _val,IList<Node> _children) {
        val = _val;
        children = _children;
    }
}

public class Solution {
    public IList<IList<int>> LevelOrder(Node root) {
        List<IList<int>> r = new List<IList<int>>();
        Queue<Node> q = new Queue<Node>();
        if (root == null)
            return r;
        q.Enqueue(root);
        while (q.Count != 0) {
            List<int> child = new List<int>();
            int n = q.Count;
            for (int i = 0; i < n; i++) {
                Node node = q.Dequeue();
                child.Add(node.val);
                foreach (Node nn in node.children){
                    q.Enqueue(nn);
                }
            }
            r.Add(child);
        }
        return r;
    }
}
}

